cmd/plugins.go

```diff
@@ -202,34 +214,44 @@ var pluginTidyCmd = &cobra.Command{
 				}
 
 				if !pluginInUse || tidyPurge {
-					if dryRun {
-						log.Infof("       plugin: [%s] can be removed", pluginName)
-						continue
-					}
-					pluginImportPath := filepath.Join(desktopCmdPath, f.Name())
-
 					// clean-up go.mod
-					pluginImportStr, _ := readPluginGoImport(pluginImportPath, pluginName)
+					pluginImportStr, err := readPluginGoImport(pluginImportPath, pluginName)
+
 					// Delete the 'replace' and 'require' import strings from go.mod.
 					// Not mission critical, if the plugins not correctly removed from
 					// the go.mod file, the project still works and the plugin is
 					// successfully removed from the flutter.Application.
 					if err != nil || pluginImportStr == "" {
 						log.Warnf("Couldn't clean the '%s' plugin from the 'go.mod' file. Error: %v", pluginName, err)
-					} else {
-						fileutils.RemoveLinesFromFile(filepath.Join(build.BuildPath, "go.mod"), pluginImportStr)
+					} else if err = mod.DropRequire(pluginImportStr); err != nil {
+						log.Warnf("failed remove %s from %s/go.mod: %v", pluginImportStr, build.BuildPath, err)
 					}
 
 					// remove import file
-					err = os.Remove(pluginImportPath)
-					if err != nil {
-						log.Warnf("Couldn't remove plugin %s: %v", pluginName, err)
-						continue
+					if !dryRun {
+						if err = os.Remove(pluginImportPath); err != nil {
+							log.Warnf("Couldn't remove plugin %s: %v", pluginName, err)
+							continue
+						}
 					}
 					log.Infof("       plugin: [%s] removed", pluginName)
 				}
 			}
 		}
+
+		if dryRun {
+			if s, err := modx.Print(mod); err == nil {
+				log.Infof("modified go.mod:\n%s", s)
+			} else {
+				log.Warnf("failed to print updated go.mod: %v", err)
+			}
```

Comment:
Please put the `modx.Print(..)` call and assignments to a separate line, and handle error first.

Reply:
When modx.Print(..) failed, should that be concerned a fatal error?

Reply:
sure it can be.